addons/mail/models/mail_thread.py

```diff
@@ -831,16 +831,20 @@ def _routing_check_route(self, message, message_dict, route, raise_exception=Tru
                 obj = self.env[alias.alias_parent_model_id.model].browse(alias.alias_parent_thread_id)
             else:
                 obj = self.env[model]
-            error_message = obj._alias_get_error_message(message, message_dict, alias)
-            if error_message:
+            error = obj._alias_get_error_message(message, message_dict, alias)
+            if error:
+                is_alias_error, error_message = error
                 self._routing_warn(
                     _('alias %(name)s: %(error)s', name=alias.alias_name, error=error_message or _('unknown error')),
                     message_id,
                     route,
                     False
                 )
-                body = alias._get_alias_bounced_body(message_dict)
-                self._routing_create_bounce_email(email_from, body, message, references=message_id)
+                if is_alias_error:
+                    self._notify_alias_invalid(alias.id, message, message_dict)
```

Comment:
I would keep it in the "route" section of the file (_notify prefix is used for notification methods like Inbox, SMS, ... -> Odoo notifications, not notifying people as a generic mean :) ) .

Reply:
Actually we could move the method directly on alias model. That way less questions on naming :D .

Reply:
Yes good idea, it is more natural. I have moved the method into Alias class and named _on_message_alias_config_error.